### ISSUE TITLE
`HostedUIProvidersCustomResourceInputs` issue after updating CLI and migrating auth resource

### ISSUE THREAD

**Issue by sayed-mohidin** [aws-amplify/amplify-cli#9893]
I am facing issues while pushing schema changes using amplify cli. I want to make few changes to the api and whenever I try to push I get the following errors.

`UPDATE_FAILED      HostedUIProvidersCustomResourceInputs Custom::LambdaCallout Wed Jan 05 2022 09:48:19 GMT+0300 (Arabian Standard Time) Received response status [FAILED] from custom resource. Message returned: See the details in CloudWatch Log Stream: 2022/01/09/[$LATEST]3a19b4a6a3b54f8a934b84e092336d8e (RequestId: b23eaa49-2cdb-482b-a72b-fe14af149fa3)`

`UPDATE_FAILED               authyoumi0da39d96        AWS::CloudFormation::Stack Wed Jan 05 2022 09:48:30 GMT+0300 (Arabian Standard Time) Embedded stack arn:aws:cloudformation:us-east-2:575248310550:stack/amplify-youmi-dev-104725-authyoumi0da39d96-1PCTJAN92L43Z/d590ebe0-6ec5-11ec-bba9-02002f4e277e was not successfully updated. Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: The following resource(s) failed to update: [HostedUIProvidersCustomResourceInputs].`

```
Following resources failed
Following resources failed

Resource Name: 2022/01/09/[$LATEST]3a19b4a6a3b54f8a934b84e092336d8e (Custom::LambdaCallout)
Event Type: update
Reason: Received response status [FAILED] from custom resource. Message returned: See the details in CloudWatch Log Stream: 2022/01/09/[$LATEST]3a19b4a6a3b54f8a934b84e092336d8e (RequestId: b23eaa49-2cdb-482b-a72b-fe14af149fa3)
URL: https://console.aws.amazon.com/cloudformation/home?region=us-east-2#/stacks/arn%3Aaws%3Acloudformation%3Aus-east-2%3A575248310550%3Astack%2Famplify-youmi-dev-104725-authyoumi0da39d96-1PCTJAN92L43Z%2Fd590ebe0-6ec5-11ec-bba9-02002f4e277e/events
```

`✖ An error occurred when pushing the resources to the cloud
🛑 An error occurred during the push operation: Resource is not in the state stackUpdateComplete`

AppID: d1ubl6yt089uw0
Region: us-east-2

It seems that the deployment of my backend has failed due to this reason and because of that I cannot make any changes to the database from the amplify console as it keeps looking for an active deployed backend.

So far me and my team have setup amplify backend, S3, IAM users and Cognito with identity pool and it was fine until we setup social signin. 

For context, my knowledge about amplify is very little so there is a high chance I might have configured something in a wrong way that caused this issue.

Its been quite a while and I still am not able to find a proper solution. Any kind of help will be highly appreciated

<img width="909" alt="Screen Shot 2022-01-05 at 10 01 33 AM" src="https://user-images.githubusercontent.com/79501416/148174390-63bc969f-bfbb-4418-803e-c75aac2160b2.png">

**Comment by josefaidt** [comment]
Hey @sayed-mohidin :wave: this appears to be a duplicate of https://github.com/aws-amplify/amplify-cli/issues/7685, which has since been fixed. Can you confirm whether you have continued to experience this error on later versions of the CLI?

**Comment by ENT108** [comment]
I confirm, the issue is still valid with CLI 7.6.22

**Comment by josefaidt** [comment]
Hey @ENT108 :wave: apologies for missing the reply here. Can you file a separate issue for that occurrence? I believe it would be beneficial to capture the current context of the project as this issue was addressed.

Closing issue due to inactivity. @sayed-mohidin if you are still experiencing this please reply back to this thread and we can re-open to investigate further 🙂

**Comment by gspeicher** [comment]
I had the same issue with `HostedUIProvidersCustomResourceInputs` failing on every push yesterday after an Amplify CLI upgrade prompted me to migrate our `auth` resource to a newer format. We ended up tracing the failure to the new `cli-inputs.json` file having a value for `hostedUIProviderMeta` that contained empty values for `authorize_scopes` and `AttributeMapping` for the `SignInWithApple` OAuth provider, which we are using for sign-in.

That value for `hostedUIProviderMeta` seems to get overwritten/broken any time you step through OAuth configuration using `amplify auth update`, so the most desirable fix here would be for the CLI to provide a workflow to modify those settings, or at a minimum to stop nuking any changes that the developer has already made to that value in the file.

**Comment by ENT108** [comment]
I got it sorted by reviewing cloudWatch logs in detail for push jobs `HostedUIProvidersCustomResource`. Yes, `cli-inputs` had issues.

**Comment by gspeicher** [comment]
Ah, my CloudWatch logs were not so helpful. The error thrown by `HostedUIProvidersCustomResource` was a generic internal server error so required a bit of trial and error to track down.

**Comment by sayed-mohidin** [comment]
@josefaidt Running `amplify auth update` and reconfiguring the auth through the CLI fixed the issue for me. Whenever I face the issue, I repeat the above steps and that seem to fix it. Hopefully this helps anyone facing the same problem.

**Comment by gspeicher** [comment]
That's funny... for us it is the exact opposite. Running `amplify auth update` reliably nukes our value for `hostedUIProviderMeta` in `cli-inputs.json` thereby breaking all subsequent pushes. Are you able to track down what value(s) in `cli-inputs.json` are involved for you?

**Comment by ENT108** [comment]
> That's funny... for us it is the exact opposite. Running `amplify auth update` reliably nukes our value for `hostedUIProviderMeta` in `cli-inputs.json` thereby breaking all subsequent pushes. Are you able to track down what value(s) in `cli-inputs.json` are involved for you?

It also destroys all mapping attributes.

**Comment by kylekirkby** [comment]
@josefaidt - any progress on this issue? Our deployments were working fine until we decided to test Amplify Hosting on our development environment. I ran `amplify pull` locally on my machine and now we cannot push anything. I'm getting an obscure error of: 

```
UPDATE_FAILED HostedUIProvidersCustomResourceInputs Custom::LambdaCallout Tue Jun 28 2022 10:32:24 GMT+0100 (British Summer Time) Received response status [FAILED] from custom resource. Message returned: See the details in CloudWatch Log Stream
```

any ideas how we can get unblocked?

**Comment by kylekirkby** [comment]
> @josefaidt - any progress on this issue? Our deployments were working fine until we decided to test Amplify Hosting on our development environment. I ran `amplify pull` locally on my machine and now we cannot push anything. I'm getting an obscure error of:
> 
> ```
> UPDATE_FAILED HostedUIProvidersCustomResourceInputs Custom::LambdaCallout Tue Jun 28 2022 10:32:24 GMT+0100 (British Summer Time) Received response status [FAILED] from custom resource. Message returned: See the details in CloudWatch Log Stream
> ```
> 
> any ideas how we can get unblocked?

What solved this for me was running `amplify update auth`.

**Comment by kylekirkby** [comment]
OK, hitting this issue on subsequent pushes after migrating auth... any update on a fix for this?

**Comment by gspeicher** [comment]
The configuration parameters from my original auth resource were not all preserved across migration of my auth resource either. I would check the values in your newly generated `cli-inputs.json` file found in the auth resource directory (particularly the value for `hostedUIProviderMeta`) and ensure they match the expected values from pre-migration.

**Comment by kylekirkby** [comment]
@gspeicher - I'm attempting to add our custom OIDC provider details into the CF stack.

I'm sure this was working, though.

Hopefully, the Amplify team adds custom OIDC provider support soon.

It seems that any update I'm doing to the `cli-inputs.json` file, to add this custom OIDC provider, is breaking our deployments. 

I've managed to get back to a working state without our custom OIDC provider details added.

**Comment by kylekirkby** [comment]
@josefaidt - yes, the custom OIDC provider details were not being respected. We've done this by hand now. It would be awesome if custom OIDC provider support was integrated/documented.

**Comment by josefaidt** [comment]
Hey @kylekirkby glad to hear you're up and running despite the trouble! We are tracking custom/generic OIDC provider support using this feature request https://github.com/aws-amplify/amplify-cli/issues/177, please subscribe to this issue for further updates on that topic.

Folks in the thread if you are still experiencing this issue could you please submit a [new issue](https://github.com/aws-amplify/amplify-cli/issues/new/choose) with the relevant information and minimal reproduction steps?

**Comment by ucheNkadiCode** [comment]
I'm running into this exact issue. Why was this closed? Should be just delete `hostedUIProviderMeta` in the cli file?
